packages/gguf/src/gguf.ts

```diff
@@ -82,29 +82,41 @@ class RangeView {
  * Note: A good article about binary data in JS: https://javascript.info/arraybuffer-binary-arrays
  */
 
-function readVersionedSize(view: DataView, byteOffset: number, version: Version, littleEndian: boolean): bigint {
+function readVersionedSize(
+	view: DataView,
+	byteOffset: number,
+	version: Version,
+	littleEndian: boolean
+): { value: bigint; offset: number } {
```

Comment:
nit: with this change, it is possible to create a return type
```ts
interface Slice<T>{
    value: T;
    offset: number
}
```

that could be used as return type for `readVersionedSize`, `readString`, `readMetadataValue`